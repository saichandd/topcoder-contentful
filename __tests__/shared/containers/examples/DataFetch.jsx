import _ from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

jest.setMock(require.resolve('actions/examples/data-fetch'), {
  examples: {
    dataFetch: {
      fetchDataInit: _.noop,
      fetchDataDone: _.noop,
    },
  },
});

const DataFetch = require('containers/examples/DataFetch').default;

const mockState = {
  examples: {
    dataFetch: {},
  },
};

test('Matches shallow shapshot', () => {
  const container = renderer.create((
    <DataFetch
      store={{
        dispatch: () => _.noop,
        getState: () => mockState,
        subscribe: _.noop,
      }}
    />
  ));
  expect(container.toJSON()).toMatchSnapshot();
});
