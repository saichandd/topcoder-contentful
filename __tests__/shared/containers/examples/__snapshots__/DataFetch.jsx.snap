// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div>
  Initial State: no data, and not loading yet.
</div>
`;
